convert2rhel/backup.py

```diff
@@ -104,6 +108,156 @@ def restore_pkgs(self):
         self._install_removed_pkgs()
 
 
+class BackupController(object):
+    """
+    Controls backup and restore for all restorable types.
+
+    This is the second version of a backup controller.  It handles all types of things that
+    convert2rhel will change on the system which it can restore in case of a failure before the
+    Point-of-no-return (PONR).
+
+    The basic interface to this is a stack.  When a Restorable is pushed onto the stack, it is
+    backed up.  When it is popped off of the stack, it is restored.
+    """
+
+    def __init__(self):
+        self._restorables = []
+
+    def push(self, restorable):
+        """
+        Enable a RestorableChange and track it in case it needs to be restored.
+
+        :arg restorable: RestorableChange object that can be restored later.
+        """
+        if not isinstance(restorable, RestorableChange):
+            raise TypeError("`%s` is not a RestorableChange object" % restorable)
+
+        restorable.enable()
+
+        self._restorables.append(restorable)
+
+    def pop(self):
+        """
+        Restore and then return the last RestorableChange added to the Controller.
+
+        :returns: RestorableChange object that was last added.
+        :raises IndexError: If there are no RestorableChanges currently known to the Controller.
+        """
+        restorable = self._restorables.pop()
+        restorable.restore()
+
+        return restorable
+
+    def pop_all(self):
+        """
+        Restores all RestorableChanges known to the Controller and then returns them.
+
+        :returns: List of RestorableChange objects that were known to the Controller.
+        :raises IndexError: If there are no RestorableChanges currently known to the Controller.
+
+        After running, the Controller object will not know about any RestorableChanges.
+        """
+        restorables = self._restorables
+
+        if not restorables:
+            raise IndexError("pop from empty list")
```

Comment:
I've changed the error message both here and in `BackupController.pop()`.  I also added an exception handler where this is called in `main.py` to log this as `info` rather than letting it propogate as an error.  If rollback is somehow triggered so early that no changes have been registered with `BackupController`, then it shouldn't be an error.